api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -193,4 +202,21 @@ message RateLimitResponse {
   // - :ref:`envoy.filters.network.ratelimit <config_network_filters_ratelimit_dynamic_metadata>` for network filter.
   // - :ref:`envoy.filters.thrift.rate_limit <config_thrift_filters_rate_limit_dynamic_metadata>` for Thrift filter.
   google.protobuf.Struct dynamic_metadata = 6;
+
+  // Quota is available for a request if its whole descriptor set has cached quota available.
+  // This is a union of all descriptors in the descriptor set. Clients can use the quota for future matches if and only if the descriptor set matches what was sent in the request that originated this response.
+  //
+  // If quota is available, a RLS request will not be made and the quota will be reduced by 1 for all matching descriptor sets.
+  //
+  // If there is not sufficient quota, there are three cases:
+  // 1. A cached entry exists for a RLS descriptor set that is out-of-quota, but not expired.
+  //    In this case, the request will be treated as OVER_LIMIT.
+  // 2. The RLS descriptor set lacks a cached entry, this will trigger a new RLS request,
+  //    When the result is returned, a single unit will be consumed from the quota for all
+  //    matching descriptor sets.
+  //    If the server did not provide a quota, such as the quota message is empty for some of
+  //    the descriptor sets, then the request admission is determined by the
+  //    :ref:`overall_code <envoy_v3_api_field_service.ratelimit.v3.RateLimitResponse.overall_code>`.
```

Comment:
This part I think needs to be updated similarly to avoid talking about "some descriptor sets"